nibabel/wrapstruct.py

```diff
@@ -536,8 +536,18 @@ def __str__(self):
 
         def _getter(obj, key):
             try:
-                return obj.get_value_label(key)
+                val = obj.get_value_label(key)
             except ValueError:
-                return obj[key]
+                val = obj[key]
+            # Convert a dtype('S1') numpy string to a Python str, if possible
+            if isinstance(val, np.ndarray) and val.dtype.type == np.bytes_:
+                try:
+                    tmp = val.tolist().decode()
+                    # Do not return unicode objects (Python 2)
+                    if isinstance(tmp, str):
```

Comment:
Actually, I was wrong. That `if` statement never evaluated `True` because the `.decode()` call errors out in Python 2 (would need to be `.decode('utf-8')`). `'%s' % np.asarray(u'xyz')` produces a `UnicodeEncodeError` in Python 2, so using `.decode('utf-8')` is a poor choice, anyway.

Happily, if I just remove the `if` branch, Python 3 will still decode Unicode, Python 2 will still error out and return as normal. It's not the most elegant thing, so I'm open to other options, or I can just document why it works.